### ISSUE TITLE
Not Satisfied with Kata - OK. Why?

### ISSUE THREAD

**Issue by dinglemouse2250** [Codewars/codewars.com#1138]
Suggest requiring the "Not Satisfied" voters to give a *reason* (which could appear in the discourse an issue)

Because how can an author (e.g. me) know how to improve a Katas if an unsatisfied user does not bother to say what the perceived problem is?

Sometimes I think these satisfaction votes are little more than just an indication of the user state of mind.

**Comment by docgunthrop** [comment]
More than likely there will be at least some number of users who would prefer not to have it as suggested. For any number of reasons, a user may want to keep their votes anonymous, especially when it comes to the negative votes they've posted.

Perhaps some changes to the voting system instead. For example, something like:

- users who complete a kata at a higher kyu rank than the user's rank cannot downvote a kata (e.g. 8kyu user cannot downvote a 6kyu kata), except when user is at least 4 kyu rank
- when users downvote, allow them to comment anonymously.
- when additional users complete the same kata and find a comment congruent to their own sentiment, they can simply upvote the comment. Each comment shows the number of upvotes it received.
- kata authors can act upon the comments by making revisions to the kata, then posting a response to each comment.
- if the user's issue is resolved by the author within a specific time window, the user who posted the comment as well as all others who upvoted the same comment get a notification.
- each of these users can then decide if they're satisfied with the changes and change their original vote, or leave it as is otherwise.

Not sure if this is feasible due to it possibly being very time-consuming (possibly breaking some things) as well as increasing burden on CW resources.

As for the user state of mind aspect, I'm not sure if there's a possible solution to that.

**Comment by dinglemouse2250** [comment]
Well ... back with this gripe/suggestion again. 

It is very frustrating getting "Not Satisfied" votes for what I consider perfectly good quality Kata that I've spent a lot of time on. I think the "Not Satisfied" votes should have a mandatory "reason" text field to go with it. 

I am don't care if there is an "anonymous" checkbox if the user wants to hide behind it. But without knowing a reason how can the issue be addressed? And if they just enter a garbage reason like "xxx", then at least I will know the vote just some vindictive nones and not a real Kata problem at all.

**Comment by JomoPipi** [comment]
If they're the first downvote, then they probably won't be anonymous anyway.

**Comment by JomoPipi** [comment]
I just spent 3 days fixing and polishing a kata just to have someone downvote it from 100% to 50% without giving any reason.

I'm fucking pissed. I was going to go straight to fixing an older beta kata of mind after this, but this really kills motivation.

Clicking "not satisfied" should open a little dialog that prompts you to write a message offering constructive feedback.

Many kata authors would appreciate this.

Please consider this issue again, @kazk!

**Comment by Voileexperiments** [comment]
I really hate the suggestions by various people over the years that for some reasons it should be mandated to have everyone who wants to vote "not satisfied" perform the equivalent of "writing a 1000-word essay why the downvote is justified and have it approved by 3 reviews".

Beta katas are filled with hot garbage (Hell, even *approved katas* are filled with them), and there are even more kata authors that does dubious practices in their katas, either due to inexperience or ignorance, and they're just left there not fixed forever. 
 The only thing that allows me, or anyone, to defend ourselves from all the shitty katas corroding the sanity and giving a constantly bad experience is the "not satisfied" button. If anyone's gonna tell me I have to prove the downvote I'm going tell them to fxck off and fend for themselves.

Seriously, have *you* done most of the katas on CW, and be so frustrated by the overall experience you've never had a single day of satisfied CW experience? That's me. Until you've gone through this stage I'm not gonna treat your opinions as qualified, period.

**Comment by JomoPipi** [comment]
@Voileexperiments, I bet you had good days when you still had approved 100% satisfaction rated kata left to solve, didn't you?
I might not have been where you have, but I've done not-so-great kata and they usually have a lower satisfaction percentage than usual. So it makes sense to prefer kata with higher ratings. Lower ratings, especially in beta, mean less people are going to be willing to try to solve it, so as to avoid that frustrating experience you talk about.

Having a troll/upset person downvote a kata that you worked hard on is a far more frustrating experience than having leave a little note when giving a satisfaction rating. No one said anything about 1000 words. One sentence - less than a fraction of a typical solution - should suffice.

If not a required message, then what is a solution you're willing to consider?

**Comment by Voileexperiments** [comment]
> I bet you had good days when you still had approved 100% satisfaction rated kata left to solve, didn't you?

No. I already said many times in gitter chat how I'm finding an average of 1 glaring issue per freshly approved kata. Please don't assume my experience.

> Lower ratings mean less people are going to be willing to try to solve it

Are you sure? There are many katas with 100% satisfaction rating but still has <50 solves even after a year. (The trainer is partly to blame about this, because it keeps suggesting the same katas with lots of solves.) And it isn't rare for me to find glaring issues in such katas too, simply because *nobody bothered to raise one*.

> Having a troll/upset person downvote a kata that you worked hard on is a far more frustrating experience than having leave a little note when giving a satisfaction rating

Are we *that* low we're setting up strawmen to attack already? Why do we keep bringing up hypothetical "trolls"/"grumpy dudes" whenever this topic is discussed? Is facing downvotes gracefully *that* hard for everyone involved in the discussion?

I'm not a fan of treating people like snowflakes because they might feel hurt. If anything, I wish people in CW would care about *my* feeling a bit more, so I won't feel like my time spent on CW has gone to waste every time I come back to solve some katas.

> If not a required message, then what is a solution you're willing to tolerate?

I reserve my right to not engage with kata authors I know will be a waste of everybody's time, while still exercising my right to vote and have it count as any other vote.

Also, frankly speaking, there are no "problems" to speak with, so no solution is needed.

**Comment by hobovsky** [comment]
I wanted to chime in and say how mandatory comment will just increase visit counters of lorem ipsum generators and that if anyone wants to, they can leave constructive feedback even now and for some reason people just don't bother, but Voile was faster to answer so just let me quote and upvote something what almost exactly reflects my thoughts:

> Why do we keep bringing up hypothetical "trolls"/"grumpy dudes" whenever this topic is discussed? Is facing downvotes gracefully _that_ hard for everyone involved in the discussion?

👍

**Comment by JomoPipi** [comment]
@Voileexperiments, 
Sorry for making assumptions. I often tend to think people are more similar to me than they really are.
In the end I'm just trying to argue for my case and make the downvote go away. 

And I realized I would probably look like a snowflake before engaging with this thread, but I don't care.
This is the worst time I've ever had on Codewars. I'm not just going to be quiet about it.

> Is facing downvotes gracefully that hard for everyone involved in the discussion?

Usually, no. But this one just sucked. The kata had 100% satisfaction, and I just spent 3 days going hard on it, fixing it, writing tests, and polishing it. I was just really excited for other people to solve it. 

Then someone comes and downvotes it without saying anything, and now it's at 50%. I was really excited about the kata, and this kind of just shit on me. Also, the reason why I mentioned "troll" is because I heard that the user is a serial downvoter. 

>  I wish other PUs in CW would care about my feeling a bit more, so I won't feel like my time spent on CW has gone to waste every time I come back to solve some katas.

I *do* care, man. 

- I try to craft my kata just like the ones I really enjoy. 
- I try to make my descriptions as clear and concise as possible. 
- I've learned how to improve my kata authoring from listening to others' feedback.
- I try to make sure that the problem is *fun*, and that nothing about it is annoying/quirky. 
- I don't stop working on fixing my kata when I know it has a bug - it needs to be fixed before anyone notices. I cringe at the thought of people noticing something wrong with my katas.

Would you not also be upset?

**Comment by Voileexperiments** [comment]
> Also, the reason why I mentioned "troll" is because I heard that the user is a serial downvoter.

Then instead of vague finger pointing and witch hunting, you should contain the mods/admins about possible serial downvoting and let them resolve that; otherwise we're gonna have a thousand threads of people doing the same, and the voting process will die in a fire from all the speculations and accusations.

I don't think feeling bad about something is a bad thing (I feel bad and vent about my CW experience more than others), but the misery that you encountered aside, threads like this keep being opened as far as I joined CW a long time ago, and these threads never end up being constructive since it's all angry venting and finger pointing. You're basically preaching to the choir about the infuriating insanity known as the beta process here ;-)

**Comment by JomoPipi** [comment]
> Then instead of vague finger pointing and witch hunting, you should contain the mods/admins about possible serial downvoting and let them resolve that

To be fair, I heard it from a mod, but then admin said that it might not really be the case.

The same user also completed my most recent kata (not the one from this complaint, but shortly after) but didn't downvote it. There also wasn't a 100% satisfaction rating he could ruin.
If he could just answer me about why he did that, then it would make things better.

Well, I hope to hear of any possible improvements.
But if my kata stays in beta for longer than 3 years, I'm deleting it and reopening it as another one...
